activestorage/app/models/active_storage/blob.rb

```diff
@@ -110,6 +110,9 @@ def text?
   # Hiding the +service_url+ behind a redirect also gives you the power to change services without updating all URLs. And
   # it allows permanent URLs that redirect to the +service_url+ to be cached in the view.
   def service_url(expires_in: service.url_expires_in, disposition: :inline, filename: self.filename, **options)
+    filename ||= self.filename
+    filename = ActiveStorage::Filename.new(filename) if filename.is_a?(String)
+
```

Comment:
Let’s extract `ActiveStorage::Filename.wrap` and call it below:

```ruby
class ActiveStorage::Filename
  def self.wrap(filename)
    filename.kind_of?(self) ? filename : new(filename)
  end
end
```

Reply:
done